### PR TITLE
Added new 2016 Wedding skills

### PR DESCRIPTION
**Pull request by aleos89**
* **Addressed Issue(s)**: None

* **Server Mode**: Renewal

* **Description of Pull Request**: 
  * Adds Call All Family, One Forever, and Cheer Up skills to the renewal databases.
  * Removed Enlarge Weight Limit R skill from a few job skill trees as it's not needed to be defined there.
Thanks to @Rytech2!

- [x] Add a SQL upgrade file to give the new Cheer Up skill to previously adopted babies.

### CODE REVIEW

File: src/map/skill.c [hunk 1]
```diff
@@ -8356,6 +8358,75 @@ int skill_castend_nodamage_id (struct block_list *src, struct block_list *bl, ui
...
+			if ((map_flag_gvg2(bl->m) || map[bl->m].flag.battleground)) { // No reviving in WoE grounds!
+				clif_skill_fail(sd, skill_id, USESKILL_FAIL_LEVEL, 0);
+				break;
+			}
+			if (status_isdead(bl)) {
+				int per = 30, sper = 0;
+
+				if (battle_check_undead(tstatus->race, tstatus->def_ele))
+					break;
+				if (tsc && tsc->data[SC_HELLPOWER])
```
Comment (Jeybla) [inline src/map/skill.c]: I'd suggest a `pc_revive_possible` function where those things are checked to reduce redundancy.

**Comment by Jeybla** [comment]
Is `WE_CALLALLFAMILY` allowed in BG, WoE and instances?

File: src/map/skill.c [hunk 2]
```diff
@@ -8356,6 +8358,75 @@ int skill_castend_nodamage_id (struct block_list *src, struct block_list *bl, ui
...
+			if (!p_sd && !c_sd) { // Fail if no family members are found
+				clif_skill_fail(sd, skill_id, USESKILL_FAIL_LEVEL, 0);
+				map_freeblock_unlock();
+				return 1;
+			}
+			if ((map_flag_gvg2(bl->m) || map[bl->m].flag.battleground)) { // No reviving in WoE grounds!
+				clif_skill_fail(sd, skill_id, USESKILL_FAIL_LEVEL, 0);
+				break;
+			}
+			if (status_isdead(bl)) {
```
Comment (Lemongrass3110) [inline src/map/skill.c]: No fail message if the target isnt dead?

File: src/map/skill.c [hunk 3]
```diff
@@ -8356,6 +8358,75 @@ int skill_castend_nodamage_id (struct block_list *src, struct block_list *bl, ui
...
+	case WE_ONEFOREVER:
+		if (sd) {
+			struct map_session_data *p_sd = pc_get_partner(sd);
+			struct map_session_data *c_sd = pc_get_child(sd);
+
+			if (!p_sd && !c_sd) { // Fail if no family members are found
+				clif_skill_fail(sd, skill_id, USESKILL_FAIL_LEVEL, 0);
+				map_freeblock_unlock();
+				return 1;
+			}
```
Comment (Lemongrass3110) [inline src/map/skill.c]: Add a dstsd check here? If it is not p_sd or c_sd cancel as well? Might be better than doing it at the bottom.

File: src/map/skill.c [hunk 4]
```diff
@@ -8356,6 +8358,75 @@ int skill_castend_nodamage_id (struct block_list *src, struct block_list *bl, ui
...
+			if (status_isdead(bl)) {
+				int per = 30, sper = 0;
+
+				if (battle_check_undead(tstatus->race, tstatus->def_ele))
+					break;
+				if (tsc && tsc->data[SC_HELLPOWER])
+					break;
+				if (map[bl->m].flag.pvp && dstsd && dstsd->pvp_point < 0)
+					break;
+				if (dstsd && dstsd->special_state.restart_full_recover)
```
Comment (Lemongrass3110) [inline src/map/skill.c]: Remove dstsd nullpo check

File: src/map/skill.c [hunk 5]
```diff
@@ -8356,6 +8358,75 @@ int skill_castend_nodamage_id (struct block_list *src, struct block_list *bl, ui
...
+					per = sper = 100;
+				if (dstsd && (dstsd->status.char_id == sd->status.partner_id || dstsd->status.char_id == sd->status.child) && status_revive(bl, per, sper)) // Only family members can be revived
```
Comment (Lemongrass3110) [inline src/map/skill.c]: Remove dstsd nullpo check and target compare

File: src/map/skill.c [hunk 6]
```diff
@@ -8356,6 +8358,75 @@ int skill_castend_nodamage_id (struct block_list *src, struct block_list *bl, ui
...
+	case WE_CHEERUP:
+		if (sd) {
+			struct map_session_data *f_sd = pc_get_father(sd);
+			struct map_session_data *m_sd = pc_get_mother(sd);
+
+			if (!f_sd && !m_sd) { // Fail if neither parent is found
+				clif_skill_fail(sd, skill_id, USESKILL_FAIL_LEVEL, 0);
+				map_freeblock_unlock();
+				return 1;
+			}
```
Comment (Lemongrass3110) [inline src/map/skill.c]: Again do dstsd check and sd compares here and not at the bottom to cancel as fast as possible.

**Pull request merged**